From 3ceb8b50272edabbbd179501a08e9f39aa164aaf Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Fri, 1 Aug 2014 21:45:34 -0400
Subject: [PATCH 1/6] Don't free the hash if another one exists

Other dissectors may still have references. This is a quick hacky fix that adds
a (very small) memory leak since the real fix in master is way too big to
backport.

Conflicts:
	epan/dissectors/packet-rtp.c

Bug:9920
Change-Id: I6501b0e8a9616b0ddc72c4e7a7f9a9ebe145b7c5
Reviewed-on: https://code.wireshark.org/review/3342
Reviewed-by: Michael Mann <mmann78@netscape.net>
Reviewed-on: https://code.wireshark.org/review/4422
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-rtp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/epan/dissectors/packet-rtp.c b/epan/dissectors/packet-rtp.c
index 472c62a..3375372 100644
--- a/epan/dissectors/packet-rtp.c
+++ b/epan/dissectors/packet-rtp.c
@@ -657,8 +657,9 @@ srtp_add_address(packet_info *pinfo, address *addr, int port, int other_port,
 	/*
 	 * Update the conversation data.
 	 */
-	/* Free the hash if already exists */
-	rtp_free_hash_dyn_payload(p_conv_data->rtp_dyn_payload);
+	/* Free the hash if a different one already exists */
+	/* if (p_conv_data->rtp_dyn_payload != rtp_dyn_payload) */
+	/* 	rtp_free_hash_dyn_payload(p_conv_data->rtp_dyn_payload); */
 
 	g_strlcpy(p_conv_data->method, setup_method, MAX_RTP_SETUP_METHOD_SIZE+1);
 	p_conv_data->frame_number = setup_frame_number;
-- 
2.1.1

